[WARNING] Unable to parse unknown SET query from client. Setting lock_hostgroup. Please report a bug for future enhancements:set @master_binlog_checksum= @@global.binlog_checksum

As above, noticeable in the log files, it's coming from Debezium (which probably should not be going through ProxySQL anyway).

ProxySQL version 3.0.1-420-g2c26a42, codename Truls

Running from a container FROM proxysql/proxysql:3.0.1

adding proxysql log for failed run
```
2025-08-06 09:23:04 [INFO] Using OpenSSL version: OpenSSL 3.5.1 1 Jul 2025
2025-08-06 09:23:04 [INFO] Using config file /var/lib/jenkins/scripts/proxysql_repl_tests/conf/proxysql/proxysql.cnf
2025-08-06 09:23:04 [INFO] Using jemalloc with MALLOC_CONF: config.xmalloc:1, lg_tcache_max:16, opt.prof_accum:1, opt.prof_leak:1, opt.lg_prof_sample:20, opt.lg_prof_interval:30, rc:0
2025-08-06 09:23:04 [INFO] Current RLIMIT_NOFILE: 1024
2025-08-06 09:23:04 main.cpp:2359:main(): [ERROR] Current RLIMIT_NOFILE is very low: 1024 .  Tune RLIMIT_NOFILE correctly before running ProxySQL
2025-08-06 09:23:04 main.cpp:2366:main(): [WARNING] Automatically setting RLIMIT_NOFILE to 102400
2025-08-06 09:23:04 [INFO] No SSL keys/certificates found in datadir (/var/lib/jenkins/workspace/ProxySQL-Automated-Build-Testing/ci_infra_logs/repl_infra/proxysql). Generating new keys/certificates.
2025-08-06 09:23:04 [INFO] ProxySQL version 3.0.1-197-gd49a96d
...
2025-08-06 09:24:15 MySQL_Session.cpp:4584:handler_minus1_LogErrorDuringQuery(): [WARNING] Error during query on (2,mysql1.proxysql_repl_tests,3306,33): 1064, You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'LOG STATUS' at line 1
2025-08-06 09:24:15 MySQL_Session.cpp:4584:handler_minus1_LogErrorDuringQuery(): [WARNING] Error during query on (2,mysql1.proxysql_repl_tests,3306,33): 1064, You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'LOG STATUS' at line 1
2025-08-06 09:24:15 MySQL_Session.cpp:4584:handler_minus1_LogErrorDuringQuery(): [WARNING] Error during query on (2,mysql1.proxysql_repl_tests,3306,34): 1064, You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'LOG STATUS' at line 1
2025-08-06 09:24:17 MySQL_Session.cpp:7890:unable_to_parse_set_statement(): [WARNING] Unable to parse unknown SET query from client 172.20.0.7:55290. Setting lock_hostgroup. Please report a bug for future enhancements:set @master_binlog_checksum= @@global.binlog_checksum
2025-08-06 09:24:17 [INFO] Received command MYSQL_COM_BINLOG_DUMP_GTID from client 172.20.0.7:55290 . Changing session fast_forward to true
2025-08-06 09:24:17 [INFO] Failed to switch to fast-forward; session connection lacks the required capabilities   hostgroup=2 client_flags=557572 server_capabilities=3254779903
2025-08-06 09:24:22 MySQL_Session.cpp:4584:handler_minus1_LogErrorDuringQuery(): [WARNING] Error during query on (0,mysql1.proxysql_repl_tests,3306,18): 1295, This command is not supported in the prepared statement protocol yet
2025-08-06 09:24:22 MySQL_Session.cpp:4584:handler_minus1_LogErrorDuringQuery(): [WARNING] Error during query on (0,mysql1.proxysql_repl_tests,3306,19): 1295, This command is not supported in the prepared statement protocol yet
2025-08-06 09:24:22 MySQL_Session.cpp:4584:handler_minus1_LogErrorDuringQuery(): [WARNING] Error during query on (0,mysql1.proxysql_repl_tests,3306,29): 1295, This command is not supported in the prepared statement protocol yet
2025-08-06 09:24:22 MySQL_Session.cpp:4584:handler_minus1_LogErrorDuringQuery(): [WARNING] Error during query on (0,mysql1.proxysql_repl_tests,3306,20): 1295, This command is not supported in the prepared statement protocol yet
2025-08-06 09:24:22 [INFO] Starting new mysql event log file /var/lib/jenkins/workspace/ProxySQL-Automated-Build-Testing/ci_infra_logs/repl_infra/proxysql/query.log.00000002
2025-08-06 09:25:17 MySQL_Session.cpp:7890:unable_to_parse_set_statement(): [WARNING] Unable to parse unknown SET query from client 172.20.0.7:41784. Setting lock_hostgroup. Please report a bug for future enhancements:set @master_binlog_checksum= @@global.binlog_checksum
2025-08-06 09:25:17 [INFO] Received command MYSQL_COM_BINLOG_DUMP_GTID from client 172.20.0.7:41784 . Changing session fast_forward to true
2025-08-06 09:25:17 [INFO] Failed to switch to fast-forward; session connection lacks the required capabilities   hostgroup=2 client_flags=557572 server_capabilities=3254779903
```

full log attached.
[proxysql.log](https://github.com/user-attachments/files/21615121/proxysql.log)

--- a/lib/mysql_connection.cpp
+++ b/lib/mysql_connection.cpp
@@ -738,8 +738,29 @@ bool MySQL_Connection::match_ff_req_options(const MySQL_Connection *c) {
 
 	// Only required to be checked for fast_forward sessions
 	if (frontend->myds && frontend->myds->sess->session_fast_forward) {
-		return (frontend->options.client_flag & CLIENT_DEPRECATE_EOF) ==
+		bool ret = (frontend->options.client_flag & CLIENT_DEPRECATE_EOF) ==
 				(backend->mysql->server_capabilities & CLIENT_DEPRECATE_EOF);
+		if (ret == false) {
+			if (frontend->myds && frontend->myds->PSarrayIN) {
+				PtrSizeArray * PSarrayIN = frontend->myds->PSarrayIN;
+				if (PSarrayIN->len == 1) {
+					PtrSize_t pkt = PSarrayIN->pdata[0];
+					if (pkt.size >= 5) {
+						unsigned char c = *reinterpret_cast<unsigned char*>(static_cast<char*>(pkt.ptr) + 4);
+						switch ((enum_mysql_command)c) {
+							case _MYSQL_COM_BINLOG_DUMP:
+							case _MYSQL_COM_BINLOG_DUMP_GTID:
+							case _MYSQL_COM_REGISTER_SLAVE:
+								ret = true;
+								break;
+							default:
+								break;
+						};
+					}
+				}
+			}
+		}
+		return ret;
 	} else {
 		return true;
 	}

